apps/onlinereports/pages/search.tsx

```diff
@@ -0,0 +1,161 @@
+import {Box, css, Skeleton, styled} from '@mui/material'
+import {
+  alignmentForTeaserBlock,
+  ApiV1,
+  articleToTeaser,
+  useWebsiteBuilder
+} from '@wepublish/website'
+import {useRouter} from 'next/router'
+import {FormEvent, useEffect, useMemo, useRef, useState} from 'react'
+import {OnlineReportsBaseTeaser} from '../src/onlinereports-base-teaser'
+
+export const SearchForm = styled('form')`
+  display: grid;
+  grid-template-columns: auto auto;
+  gap: ${({theme}) => theme.spacing(1)};
+`
+
+export const SearchResultsContainer = styled('div')`
+  display: grid;
+  grid-template-columns: 1fr;
+  gap: ${({theme}) => theme.spacing(2)};
+
+  ${({theme}) => css`
+    ${theme.breakpoints.up('sm')} {
+      grid-template-columns: repeat(2, 1fr);
+    }
+  `}
+
+  ${({theme}) => css`
+    ${theme.breakpoints.up('lg')} {
+      grid-template-columns: repeat(3, 1fr);
+    }
+  `}
+`
+
+const ITEMS_PER_PAGE = 12
+
+export default function Search() {
+  const [searchTerm, setSearchTerm] = useState<string | undefined>(undefined)
+  const [page, setPage] = useState<number>(1)
+
+  const router = useRouter()
+  const {
+    elements: {Button, TextField, Pagination}
+  } = useWebsiteBuilder()
+
+  const [searchRequest, {error, data, loading}] = ApiV1.usePhraseLazyQuery()
+
+  function search(e: FormEvent) {
+    e?.preventDefault()
+    // reset page
+    setPage(1)
+    router.push({
+      pathname: '/search',
+      query: {q: searchTerm}
+    })
+  }
+
+  function changePage(newPage: number) {
+    setPage(newPage)
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth'
+    })
+  }
+
+  const searchQuery: string | undefined = (router.query.q as string) || undefined
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus()
+    }
+  })
```

Comment:
use effect is missing dependencies. This should use ref callbacks ideally. I know it's in the other search page as well but it's also wrong there.